src/tests/unit/tests/background/browser-permissions-tracker.test.ts

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { allUrlAndFilePermissions, BrowserPermissionsTracker } from 'background/browser-permissions-tracker';
+import { Interpreter } from 'background/interpreter';
+import { BrowserAdapter } from 'common/browser-adapters/browser-adapter';
+import { Logger } from 'common/logging/logger';
+import { Message } from 'common/message';
+import { Messages } from 'common/messages';
+import { isFunction } from 'lodash';
+import { IMock, It, Mock, Times } from 'typemoq';
+
+describe('BrowserPermissionsTracker', () => {
+    let testSubject: BrowserPermissionsTracker;
+    let browserAdapterMock: SimulatedBrowserAdapter;
+    let interpreterMock: IMock<Interpreter>;
+    let loggerMock: IMock<Logger>;
+
+    beforeEach(() => {
+        interpreterMock = Mock.ofType<Interpreter>();
+        loggerMock = Mock.ofType<Logger>();
+        browserAdapterMock = createBrowserAdapterMock();
+
+        testSubject = new BrowserPermissionsTracker(browserAdapterMock.object, interpreterMock.object, loggerMock.object);
+    });
+
+    describe('initialize', () => {
+        it('registers the expected listeners', async () => {
+            await testSubject.initialize();
+
+            browserAdapterMock.verify(m => m.addListenerOnPermissionsAdded(It.is(isFunction)), Times.once());
+            browserAdapterMock.verify(m => m.addListenerOnPermissionsRemoved(It.is(isFunction)), Times.once());
+        });
+    });
+
+    describe('onPermissionsAdded', () => {
+        it.each([true, false])(
```

Comment:
The tests look good! However, I do notice a lot of overlap in these scenarios that you can likely refactor.

An example would be a "getExpectedMessage" function with the expected browserPermissions value as a parameter. Something similar might be done with the browser adapter setup.